### PR TITLE
Implement PlayQueue

### PR DESCRIPTION
**Pull request by raimis**
- [x] Implement PlayQueue
  - [x] Execution
  - [x] Logging

### CODE REVIEW

File: htmd/queues/playqueue.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+            elif status in (4, 5): # Completed or errored
+                pass
+            else:
+                raise ValueError('Unknown job status')
+
+        return counter
+
+    def retrieve(self):
+
+        for jobID, dir_ in list(self._dirs.items()):
```
Comment (stefdoerr) [inline htmd/queues/playqueue.py]: no need to wrap in `list()`

Comment (raimis) [reply htmd/queues/playqueue.py]: Bare in mind, `self._dirs.items()` gives an iterator, not a list. So ` self._dirs.pop(jobID)` modifies iterable used by that iterator in the loop, which is not allowed.

Comment (stefdoerr) [reply htmd/queues/playqueue.py]: ah kk

**Comment by raimis** [comment]
> I don't see this working under most realistic scenarios.

It is already work in *Parameterize* App: https://github.com/Acellera/playmolecule-parameterize/pull/18

> If you close the python session or it dies you lose the job IDs and you cannot get them back. Also this forbids it's usage in asynchronous stuff like adaptive if we ever want to.

The other queues have the same characteristic.

> I would do it with a local DB (sqlite probably since it's best integrated in python) which keeps track of submitted jobs and folders.

It is just a thin wrapper of *PM* API. Why do we need that complexity?

**Comment by stefdoerr** [comment]
That's not exactly true. SLURM and LSF both poll the server for currently running jobs with a given project name so the "inprogress" method works without need to track the jobs locally. Since I don't know if PM keeps track of jobs on a server, if it does, then we should also poll the server for those, otherwise keep track locally of the job ids

How does it work if you restart the python job?

**Comment by raimis** [comment]
> That's not exactly true. SLURM and LSF both poll the server for currently running jobs with a given project name so the "inprogress" method works without need to track the jobs locally. Since I don't know if PM keeps track of jobs on a server, if it does, then we should also poll the server for those, otherwise keep track locally of the job ids

It could be done with `playmolecule.Job.getChildren`, but I haven't managed to work. It needs some serious debugging at *PM* level. Yes, you are right, now it works more like `LocalQueue` than `SlurmQueue`.

> How does it work if you restart the python job?

At the moment, *Parameterize* can only recover completed jobs, running jobs will be lost with any queues.

**Comment by stefdoerr** [comment]
okay if we want it to work from the PM side then I will try to fix that once I'm back full time. For the moment we can leave it as is, just maybe add a TODO in the inprogress that we should make it work from the PM side.

**Comment by raimis** [comment]
Done! Any more comments?

**Pull request merged**